cborg/src/Codec/CBOR/Write.hs

```diff
@@ -388,6 +394,10 @@ stringLenMP =
 stringBeginMP :: P.BoundedPrim ()
 stringBeginMP = constHeader 0x7f
 
+utf8ByteArrayMP :: BA.ByteArray -> B.Builder
+utf8ByteArrayMP t =
+    P.primBounded stringLenMP (fromIntegral $ BA.sizeofByteArray t) <> BA.toBuilder t
```

Comment:
Like I mentioned with the decoders, we should clearly document if the encoder instructions are going to be valiating or not. So we should document that `encodeUtf8ByteArray` does not check that the content is valid UTF8, but expects it to be.